### ISSUE TITLE
Adding "context" of the current page to components.

### ISSUE THREAD

**Issue by joaojoyce** [measuredco/puck#476]
I'm wondering if there is any way to pass a context object to `<Puck />` and `<Render />` components. 

This context could be passed directly to render() or resolveData() methods and wouldn't need to be stored.

### Use case 1:
I want to create a `<RelatedPosts />` component, that would show the latest posts of a blog excluding the post I'm currently in.
But for that I would need resolveData() to be aware what the current post slug/id is, in order to filter out the current post.

### Use case 2:
In render() I want to show user specific data (example: favorite products, favorite artists, list of friends, etc...) that I don't have while editing.

Is there any other way to do this I'm not currently considering? 
I would be able to contribute with code if you find this useful.

Thank you for your awesome work.

**Comment by chrisvxd** [comment]
You can put your provider outside the Puck or Render components, and consume the context within the `render` method as normal.

Something like this - 

```tsx
const ctx = createContext();

const { Provider } = ctx;

const Editor = () => {
  return (
    <Provider value={{ blogPosts: [] }}>
      <Puck
        config={{
          components: {
            MyComponent: {
              render: () => {
                const { blogPosts } = useContext(ctx);

                return (
                  <ul>
                    {blogPosts.map((blogPost) => (
                      <li>{blogPost.title}</li>
                    ))}
                  </ul>
                );
              },
            },
          },
        }}
      />
    </Provider>
  );
};
```

I believe that should cover both of your use cases. I'm going to close this issue, but please let me know if that doesn't solve your issue!

**Comment by joaojoyce** [comment]
Thank you.

I've tried a couple of possible solutions yesterday and that was one of them. It works but it was harder to pass information to `resolveAllData` through context on server side environments.

I ended up doing a simpler thing which seems to work. Just pass the context through config. I wrap the config object inside a `getBasicConfig({_context})` function. It seems to be working.

Thank you very much for your work.
